### PR TITLE
Feature/store style

### PR DESCRIPTION
**Pull request by Raycoms**
Correctly calculate ore and saplings with leaves.

Review please

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -97,7 +97,7 @@ configFile.withReader {
 }
 
 dependencies {
-    deobfCompile "slimeknights.mantle:Mantle:1.12-${config.mantle_version}"
+    deobfCompile "slimeknights.mantle:Mantle:1.1s2-${config.mantle_version}"
```
Comment (marchermans) [inline build.gradle]: I do not think that this is correct

**Review by marchermans — CHANGES_REQUESTED** [review]
Check the build system

**Comment by Raycoms** [comment]
It's not, fixed it

2017-12-21 19:55 GMT+00:00 Marc Hermans <notifications@github.com>:

File: src/main/java/com/minecolonies/coremod/colony/ColonyManager.java [hunk 2]
```diff
@@ -89,16 +86,18 @@
     /**
      * A buffer value to be sure to be outside of the colony.
      */
-    private static final int BUFFER = 10;
+    private static final int BUFFER                    = 10;
 
     /**
      * The last colony id.
      */
     private static int topColonyId = 0;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/ColonyManager.java:94]: MAJOR Remove this unused "topColonyId" private field. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1068)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 98 issues
* ![CRITICAL][CRITICAL] 5 critical
* ![MAJOR][MAJOR] 35 major
* ![MINOR][MINOR] 55 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1437](https://github.com/Minecolonies/minecolonies/blob/f66616e255fc13e41c9b34a664e646ef0e655fa9/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1437): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1466](https://github.com/Minecolonies/minecolonies/blob/f66616e255fc13e41c9b34a664e646ef0e655fa9/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1466): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1477](https://github.com/Minecolonies/minecolonies/blob/f66616e255fc13e41c9b34a664e646ef0e655fa9/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1477): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1583](https://github.com/Minecolonies/minecolonies/blob/f66616e255fc13e41c9b34a664e646ef0e655fa9/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1583): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1697](https://github.com/Minecolonies/minecolonies/blob/f66616e255fc13e41c9b34a664e646ef0e655fa9/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1697): Missing curly brace. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00121)
1. ![MAJOR][MAJOR] [ColonyManager.java#L49](https://github.com/Minecolonies/minecolonies/blob/f66616e255fc13e41c9b34a664e646ef0e655fa9/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L49): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 32 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [ColonyManager.java#L49](https://github.com/Minecolonies/minecolonies/blob/f66616e255fc13e41c9b34a664e646ef0e655fa9/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L49): class "ColonyManager" has 51 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ColonyManager.java#L178](https://github.com/Minecolonies/minecolonies/blob/f66616e255fc13e41c9b34a664e646ef0e655fa9/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L178): Provide multiple methods instead of using "canDestroy" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [ColonyManager.java#L762](https://github.com/Minecolonies/minecolonies/blob/f66616e255fc13e41c9b34a664e646ef0e655fa9/src/main/java/com/minecolonies/coremod/colony/ColonyManager.java#L762): Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#](https://github.com/Minecolonies/minecolonies/blob/f66616e255fc13e41c9b34a664e646ef0e655fa9/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#): This file has 1.124 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**